Yoast/wordpress-seo
# Add method to retrieve posts with outdated words

## Summary


This PR can be summarized in the following changelog entry:

* Adds code change to make it possible to select posts from the database with outdated prominent words.

Adds a method to retrieve posts with out of date prominent words. It combines data from the indexables table and the posts table to retrieve the posts. All posts that don't have indexables yet are automatically included. This is because the subquery check to see that there is an indexable with an up-to-date version. Then we select all posts that don't have an indexable like this.

## Relevant technical choices:

* I've also included a change that puts `WPSEO_Replace_Vars` in the autowired classes. This is required for the changes I intend to make to premium.
* Because the query requires querying the posts table I needed to inject `wpdb` into the `Indexable_Repository` so I can get the WordPress database prefix.
* The count and find queries are so similar I've abstracted the query building part to a separate method so they can share this part.

## Test instructions

This PR can be tested by following these steps:

* Obviously the unit tests should run. This is a good test, if all is well you should be able to figure out what the code is supposed to do by reading the unit tests.
* To actually test this with the real database you can do one of two things:
    * Test it together with the PR in premium which will be here shortly.
    * Call the query method yourself from the code and then manipulate your database and inspect if the results are what you expect.

The query results should include:
* Posts that don't have an indexable
* Posts that have an indexable where the `prominent_words_version` is outdated.

The query results should NOT include:
* Posts that have an indexable and the indexable has an up-to-date `prominent_words_version`.

## Documentation

## Quality assurance


See https://github.com/Yoast/wordpress-seo-premium/issues/2532.
